src/lib/gundb/UserStorage.js

```diff
@@ -692,7 +692,9 @@ export class UserStorage {
       .get(day)
       .put(dayEventsArr.length)
 
-    const result = await Promise.all([saveAck, ack])
+    const lastBlockAck = this.saveLastBlockNumberProcessed()
+
+    const result = await Promise.all([saveAck, ack, lastBlockAck])
```

Comment:
1. ack is not a promise. you need to add .then() to make it a promise.
2. the description of the method says it returns the updated feed but it returns a Promise<ACK>, a promise which is filled when update was done.